collect_app/src/main/java/org/odk/collect/android/utilities/ContentUriHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018 Shobhit Agarwal
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.odk.collect.android.utilities;
+
+import android.content.ContentResolver;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.webkit.MimeTypeMap;
+
+import org.odk.collect.android.application.Collect;
+
+public final class ContentUriHelper {
+
+    private ContentUriHelper() {
+
+    }
+
+    public static Long getIdFromUri(Uri contentUri) {
```

Comment:
There are probably other cases where this method could be used like `InstanceProvider` where we call:
`uri.getPathSegments().get(1))` in a few places. Since now we have this helper method probably it should be used everywhere.

Reply:
Good shout will look at that.

Reply:
Actually it looks like the only uses of `getPathSegments` outside this are in `InstancesProvider`. I'll change those in the next PR after I've added test coverage.